src/rgw/driver/rados/rgw_datalog.cc

```diff
@@ -530,14 +530,18 @@ RGWDataChangesLog::start(const DoutPrefixProvider *dpp,
       throw sys::system_error{ENOTCONN, sys::generic_category(),
 			      "Unable to establish recovery watch!"};
     }
-    async::co_spawn(sg->get_executor(),
-		    watch_loop(), sg->completion());
+    async::co_spawn(co_await asio::this_coro::executor,
```

Comment:
@cbodley Is this what you had in mind? It seems to have fixed the crashes, I'm running startup/shutdown on a loop, and it hasn't crashed yet.

Reply:
looks great!